tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPReplicaTest.java

```diff
@@ -245,6 +254,142 @@ private String getText(boolean large, int i) {
+            assertNotNull(received);
+            Assert.assertEquals(getText(true, i), received.getText());
+            Assert.assertNull(received.getDeliveryAnnotation("gone"));
+         }
+         receiver.flow(1);
+         Assert.assertNull(receiver.receiveNoWait());
+
+         connection.close();
+      }
+   }
+
+
+   @Test
+   public void testReplicaNoAddressOnMessage() throws Exception {
+      server.setIdentity("targetServer");
+      server.start();
+      server_2 = createServer(AMQP_PORT_2, false);
+      server_2.setIdentity("server_2");
+
+      AMQPBrokerConnectConfiguration amqpConnection = new AMQPBrokerConnectConfiguration("test", "tcp://localhost:" + AMQP_PORT).setReconnectAttempts(-1).setRetryInterval(100);
+      AMQPMirrorBrokerConnectionElement replica = new AMQPMirrorBrokerConnectionElement().setMessageAcknowledgements(true);
+      amqpConnection.addElement(replica);
+      server_2.getConfiguration().addAMQPConnection(amqpConnection);
+
+      int NUMBER_OF_MESSAGES = 20;
+
+      server_2.start();
+      Wait.assertTrue(server_2::isStarted);
+
+      // We create the address to avoid auto delete on the queue
+      server_2.addAddressInfo(new AddressInfo("TEST").addRoutingType(RoutingType.ANYCAST).setAutoCreated(false));
+      server_2.createQueue(new QueueConfiguration("TEST").setRoutingType(RoutingType.ANYCAST).setAddress("TEST").setAutoCreated(false));
+
+      Assert.assertFalse(AssertionLoggerHandler.findText("AMQ222214"));
+
+      { // sender
+         AmqpClient client = new AmqpClient(new URI("tcp://localhost:" + AMQP_PORT_2), null, null);
+         AmqpConnection connection = addConnection(client.connect());
+         AmqpSession session = connection.createSession();
+
+         AmqpSender sender = session.createSender("TEST");
+
+         for (int i = 0; i < NUMBER_OF_MESSAGES; i++) {
+            AmqpMessage message = new AmqpMessage();
+            message.setDeliveryAnnotation("gone", "test");
+            message.setText(getText(false, i));
+            sender.send(message);
+         }
+         sender.close();
+         connection.close();
+      }
+
+      { // receiver on replica
+         AmqpClient client = new AmqpClient(new URI("tcp://localhost:" + AMQP_PORT), null, null);
+         AmqpConnection connection = addConnection(client.connect());
+         AmqpSession session = connection.createSession();
+         // Now try and get the message
+
+         AmqpReceiver receiver = session.createReceiver("TEST");
+         receiver.flow(NUMBER_OF_MESSAGES);
+         for (int i = 0; i < NUMBER_OF_MESSAGES; i++) {
+            AmqpMessage received = receiver.receive(5, TimeUnit.SECONDS);
+            assertNotNull(received);
+            Assert.assertEquals(getText(false, i), received.getText());
+            Assert.assertNull(received.getDeliveryAnnotation("gone"));
+         }
+         receiver.flow(1);
+         Assert.assertNull(receiver.receiveNoWait());
```

Comment:
It would still eventually fail if extra messages were sent.
I don't want to add a wait timeout as it would increase the time on the test suite.

this is meant to receiveNull, if it fails 1 in 10 it's already enough for my assertion here.